### ISSUE TITLE
Sparsity Table

### ISSUE THREAD

**Issue by EngineerDanny** [EngineerDanny/necromass#2]
# Public Data set
| Dataset | Sparsity (%) |
|:--------|-------------:|
| amgut1_data | 30.40 |
| amgut2_data | 34.60 |
| baxter_crc_data | 27.78 |
| crohns_data | 1.00 |
| glne007_data | 58.88 |
| global_patterns_data | 79.07 |
| esophagus_data | 43.10 |
| enterotype_data | 67.62 |
| hmp2prot_data | 14.05 |
| hmp216S_data | 12.67 |
| mixmpln_real_data | 69.82 |
| soilrep_data | 69.82 |
| ioral_data | 43.10 |

# Necromass Data set
| Dataset | Sparsity (%) |
|:--------|-------------:|
| bacteria_conservative_raw | 29.44 |
| bacteria_genus_raw | 69.48 |
| fungi_conservative_raw | 56.01 |
| fungi_genus_raw | 85.66 |
| bacteria_fungi_conservative_raw | 41.99 |
| Dec22_bacteria_conservative_r_same_raw | 35.39 |
| Dec22_fungi_conservative_r_same_raw | 58.05 |
| Dec22_bacteria_fungi_conservative_r_same_raw | 46.42 |

## Dec22_bacteria_fungi_conservative_r_same_raw
| Group | Sparsity (%) |
|:--------|-------------:|
| AllSoilM1M3 | 44.07 |
| LowMelanM1 | 46.62 |
| HighMelanM1 | 51.84 |
| LowMelanM3 | 43.00 |
| HighMelanM3 | 48.87 |

**Comment by tdhock** [comment]
looks like the sparsity of necromass data is consistent with public data (not larger as I was expecting), so that contradicts my hypothesis that increased sparsity in necromass may be causing the learning algorithm to not work (large test error rates).

**Comment by tdhock** [comment]
can you please also compute the sparsity of the updated data from Briana on Dec 22?

**Comment by EngineerDanny** [comment]
Sure. I am on it.
Although, I am thinking that maybe I should also add the sparsity of the groups(the sub-samples).
Since you used that in your computation.
Maybe your hypothesis is not wrong.
I will update on this soon

**Comment by EngineerDanny** [comment]
## Binary Classification
@tdhock  This is the results when the labels are replaced with 1(positive numbers) and 0(negative numbers) to try to address the sparsity problem.

![binary_classifier](https://github.com/EngineerDanny/necromass/assets/47421661/3b327ffb-8625-4186-9959-b578c4eba988)

**Comment by tdhock** [comment]
The X axis says MSE, is that right? 
You may consider making one plot like above with accuracy/error.
And another plot with AUC.
The legend says LogisticRegression, did you use L1 regularization? 
What did you use for features? log transform? 0/1?

**Comment by EngineerDanny** [comment]
The MSE is wrong, it is rather the accuracy (I have updated it).
Yh I will add the AUC curve next.
I used [LogisticRegressionCV](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegressionCV.html), the default penalty is L2 so this is using L2 regularization (I will change it to L1 and update).
I used the log transform features (I know it's not ideal but that's what I started with).

**Comment by tdhock** [comment]
log transform is fine.
I thought the default was L1? (not L2?)
I thought it may have been accuracy and not MSE.
In that case it looks like linear model is sometimes better than featureless, but not often. so not very encouraging. 
maybe worth adding 0/1 features + other transformations (with scaling) to the feature matrix, to see if that helps.
It could be that the necromass data are just hard to predict, so I think the next step should be to try it on all of the other data sets, to see if it does any better.

**Comment by EngineerDanny** [comment]
## Binary Classification + Regression
![output_class_reg](https://github.com/EngineerDanny/necromass/assets/47421661/0240213f-e56f-48cd-a9e9-4bd55541974f)

**Comment by EngineerDanny** [comment]
![output](https://github.com/EngineerDanny/necromass/assets/47421661/9141e3aa-ef9b-4372-ba74-5722eaf2f097)
@tdhock This is the results for the `MSE` on test set against the `# of Total Samples`
GGM just doesn't work without standard scaling.

**Comment by EngineerDanny** [comment]
@tdhock This is the result using the other public data sets.
As expected when using all the samples that there will be a considerable test error difference between Featureless and LassoCV.

![output](https://github.com/EngineerDanny/necromass/assets/47421661/c0e15945-fd49-404c-adb0-77098107827a)

**Comment by tdhock** [comment]
> ![output](https://private-user-images.githubusercontent.com/47421661/302376066-9141e3aa-ef9b-4372-ba74-5722eaf2f097.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDcyMzYxNjgsIm5iZiI6MTcwNzIzNTg2OCwicGF0aCI6Ii80NzQyMTY2MS8zMDIzNzYwNjYtOTE0MWUzYWEtZWY5Yi00MzcyLWJhNzQtNTcyMmVhZjJmMDk3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjA2VDE2MTEwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUxNmRmNTQyNjY1MmJiM2FjNDQxYzk3OWNiNTY3ZGYxNjUyYWE4OWRiMDRjM2IwNTJmMjc4YThhNmVlYTE0MjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.BHwMLAqtAj7v8ZCdHXJvIuCjn785QO430pQmCdbCKcY) @tdhock This is the results for the `MSE` on test set against the `# of Total Samples` GGM just doesn't work without standard scaling.

Can you please write your hypothesis (why do wanted to make this plot) and interpretation (does it confirm your hypothesis?)
You should probably do facet_wrap(scales="free") for this plot, so we can see the differences better. There is no reason the MSE should be on the same scale at all between the different panels.

**Comment by tdhock** [comment]
> @tdhock This is the result using the other public data sets. As expected when using all the samples that there will be a considerable test error difference between Featureless and LassoCV.
> 
> ![output](https://private-user-images.githubusercontent.com/47421661/302485732-c0e15945-fd49-404c-adb0-77098107827a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDcyMzYxNjgsIm5iZiI6MTcwNzIzNTg2OCwicGF0aCI6Ii80NzQyMTY2MS8zMDI0ODU3MzItYzBlMTU5NDUtZmQ0OS00MDRjLWFkYjAtNzcwOTgxMDc4MjdhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjA2VDE2MTEwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJlZGYxYmI4MWI3YjQ1NWI0YWYwNDVmNGI1NDU0NGVjMzI1ZjY2ODI4ZjA2NzUzOTA3MDYxY2U0YzRmZDViZTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.GPmMTJJsvMAQ6S0UW0dJTHFaFBgGYgGqWC4nCTAXRf0)

Should the y axis be "test fold" ?
Is that the original LassoCV?
Can you please add results for LogisticRegressionCV + LassoCV? (to see if predicting 0/1 helps lower test MSE, relative to LassoCV by itself?)

**Comment by EngineerDanny** [comment]
@tdhock This is the result for the FoldId against the MSE for `Featureless`, `LassoCV` and `LogisticReg+LassoCV`.
I don't think `LogisticReg+LassoCV` does better than `LassoCV`.

![output](https://github.com/EngineerDanny/necromass/assets/47421661/8b678757-3425-43ae-8f36-2a99b51966b5)

**Comment by EngineerDanny** [comment]
@tdhock This is the result for the new architecture.

![output](https://github.com/EngineerDanny/necromass/assets/47421661/c6ee4d53-cffc-41d3-a8e1-50aa88187520)

**Comment by tdhock** [comment]
looks like it is about the same as other methods in this data set. that is consistent with our hypothesis that there is little to learn in these data. have you tried the new method on the other public data?

**Comment by EngineerDanny** [comment]
@tdhock , Yh I tested it on the other data sets.
Results are consistent with that of the necromass data set.

![output](https://github.com/EngineerDanny/necromass/assets/47421661/60b47af9-8e2c-468c-a58d-8c40e052c25a)

**Comment by tdhock** [comment]
??? what are those ??? are they from public data like amgut etc?

**Comment by EngineerDanny** [comment]
Yh. They are

**Comment by tdhock** [comment]
is that result consistent with the other figure? (one panel per data set, instead of one panel per column)

**Comment by EngineerDanny** [comment]
![output](https://github.com/EngineerDanny/necromass/assets/47421661/5e8b2b7a-da38-4df9-941c-21fb3626fa80)

**Comment by tdhock** [comment]
https://www.andrewheiss.com/blog/2022/05/09/hurdle-lognormal-gaussian-brms/#normally-distributed-outcomes-with-zeros
![plot-pp-mass-bill-hurdle-log-1](https://github.com/EngineerDanny/necromass/assets/932850/f3161458-9306-4c08-ba35-f82bf465d22a)

**Comment by tdhock** [comment]
https://www.statsmodels.org/devel/examples/notebooks/generated/count_hurdle.html

**Comment by tdhock** [comment]
https://search.r-project.org/CRAN/refmans/EnvStats/html/ZeroModifiedLognormal.html

**Comment by EngineerDanny** [comment]
This is the table for the publicly available grouped-samples microbiome data:

| Data           | Source                                                      | No of Taxa | No of Samples | No of Groups | Sparsity |
|----------------|-------------------------------------------------------------|------------|---------------|--------------|----------|
| HPv13          | [PubMed 22699609](https://www.ncbi.nlm.nih.gov/pubmed/22699609) | 5830       | 3285          | 71           | 98.16%   |
| HPv35          | [PubMed 22699609](https://www.ncbi.nlm.nih.gov/pubmed/22699609) | 10730      | 6000          | 152          | 98.71%   |
| MovingPictures | [PubMed 21624126](https://www.ncbi.nlm.nih.gov/pubmed/21624126) | 22765      | 1967          | 6            | 97.06%   |
| qa10394        | [mSystems e00021-16](http://msystems.asm.org/content/1/3/e00021-16) | 9719       | 1418          | 16           | 94.28%   |
| TwinsUK        | [PubMed 25417156](https://www.ncbi.nlm.nih.gov/pubmed/25417156) | 8480       | 1024          | 16           | 87.70%   |
